tests/docker/README.md

```diff
@@ -0,0 +1,23 @@
+Test Framework
+==============
+
+This is a test framework for testing the irods_client_globus_connector.
+
+Running Tests
+-------------
+
+To run the tests first make sure you have the code that is to be tested checked out.
+
+Run the tests by executing the following in this directory:
+
+```
+docker-compose build
+docker-compose run globus
```

Comment:
Oh, or is the idea that you can leave the iRODS container running and run the Globus tests repeatedly on the same instance? That's pretty swell, if true.

Reply:
There are a couple of reasons I do this.

1. I just updated this to the following and the `docker-compose run X` allows me to target the specific instance.  So if I run `docker-compose run globus-ubuntu20` it will only start up the iRODS container (due to dependency) and the Ubuntu 20 Globus container.

```
    globus-ubuntu22:                                                                                                                                                                                       
        build:                                                                                                                                                                                             
            context: .                                                                                                                                                                                     
            dockerfile: globus.ubuntu22.Dockerfile                                                                                                                                                         
        depends_on:                                                                                                                                                                                        
            irods:                                                                                                                                                                                         
                condition: service_healthy                                                                                                                                                                 
        volumes:                                                                                                                                                                                           
            - ../../:/irods_client_globus_connector                                                                                                                                                        
    globus-ubuntu20:                                                                                                                                                                                       
        build:                                                                                                                                                                                             
            context: .                                                                                                                                                                                     
            dockerfile: globus.ubuntu20.Dockerfile                                                                                                                                                         
        depends_on:                                                                                                                                                                                        
            irods:                                                                                                                                                                                         
                condition: service_healthy                                                                                                                                                                 
        volumes:                                                                                                                                                                                           
            - ../../:/irods_client_globus_connector 
```
2. I'm not sure why but doing this doesn't interleave output from the two containers that are running (iRODS and Globus).  This makes it far easier to parse the results of the test.

Reply:
> Oh, or is the idea that you can leave the iRODS container running and run the Globus tests repeatedly on the same instance? That's pretty swell, if true.

It won't leave the container running.  If I want to leave it running I'll add a `tail -f /dev/null` just like if I used `docker-compose up`.  I'll probably accidentally check that in sooner or later.

Reply:
Oh, I see. `up` has a similar option (specifying a service which will bring up all of its dependencies as well). I'm wondering whether we ought to separate the different platforms into their own compose files rather than having each platform defined as a service in the same file. I didn't catch that before.

Reply:
That was just in my latest commit.  I kind of like it the way it is rather than duplicating it.

Plus, the ability to not have stdout from the iRODS container is a big plus with `docker-compose run` vs `docker-compose up`.